mono/metadata/assembly.c

```diff
@@ -1540,6 +1555,7 @@ load_reference_by_aname_individual_asmctx (MonoAssemblyName *aname, MonoAssembly
 	aname = mono_assembly_apply_binding (aname, &maped_name_pp);
 
 	reference = mono_assembly_loaded_full (aname, FALSE);
+	mono_assembly_pinref (reference);
```

Comment:
Why won't this race with the unloader since it's happening outside of the unload coordinating lock?